test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc

```diff
@@ -222,15 +222,30 @@ TEST_P(ProxyProtocolTest, V1Basic) {
   disconnect();
 }
 
-TEST_P(ProxyProtocolTest, AllowNoProxyProtocol) {
-  // allows request through even though it doesn't use proxy protocol
+TEST_P(ProxyProtocolTest, AllowTinyNoProxyProtocol) {
+  // allows a small request (less bytes than v1/v2 header) through even though it doesn't use proxy
+  // protocol
   envoy::extensions::filters::listener::proxy_protocol::v3::ProxyProtocol proto_config;
   proto_config.set_allow_requests_without_proxy_protocol(true);
   connect(true, &proto_config);
 
-  write("more data");
+  write("data");
 
-  expectData("more data");
+  expectData("data");
+
+  disconnect();
+}
+
+TEST_P(ProxyProtocolTest, AllowLargeNoProxyProtocol) {
+  // allows a large request (more bytes than v1/v2 header) through even though it doesn't use proxy
+  // protocol
+  envoy::extensions::filters::listener::proxy_protocol::v3::ProxyProtocol proto_config;
+  proto_config.set_allow_requests_without_proxy_protocol(true);
+  connect(true, &proto_config);
+
+  write("more data more data more data");
```

Comment:
Can you add a comment with both the v1 and v2 header prefix, so it's clear that this string is longer than both?

Reply:
opted to move these assertions into code, thought it was clearer https://github.com/envoyproxy/envoy/pull/18951/commits/a37ad43cd7cf2cedb4491a1700da4515896b27bb